src/app/pages/products/store/filters/filter.effect.ts

```diff
@@ -1,98 +1,56 @@
+    return [queryParams];
+  } else if (Array.isArray(queryParams)) {
+    return queryParams;
+  } else {
+    return [];
+  }
+}
 
 @Injectable()
 export class FilterEffects {
-  pageOpening$ = createEffect(() => {
-    return combineLatest([
-      this.actions$.pipe(ofType(FilterActions.setSelectedFilters), take(1)),
-      this.actions$.pipe(ofType(FilterActions.setSearch), take(1)),
-      this.actions$.pipe(ofType(FilterActions.setSortingOption), take(1)),
-    ])
-      .pipe(
-        tap(() => {
-          const { searchQuery, sortingOption, ...filterParams } = this.router.routerState.root.snapshot.queryParams;
-
-          if (filterParams) {
-            const selectedFilters = Object.keys(filterParams)
-              .reduce((result: Record<string, string[]>, filterField: string) => {
-                return {
-                  ...result,
-                  [filterField]: filterParams[filterField].split(','),
-                };
-              }, {});
-
-            this.store.dispatch(FilterActions.setSelectedFilters({ selectedFilters }));
-          }
-
-          if (searchQuery) {
-            this.store.dispatch(FilterActions.setSearch({ searchValue: searchQuery }));
-          }
-
-          if (sortingOption) {
-            this.store.dispatch(FilterActions.setSortingOption({ sortingOption }));
-          }
-        }),
-      );
-  }, { dispatch: false });
-
-  filtering$ = createEffect(() => {
-    return this.actions$
-      .pipe(
-        ofType(FilterActions.setSelectedFilters),
-        tap(({ selectedFilters }) => {
-          const filterParams = Object.keys(selectedFilters)
-            .reduce((result: Record<string, string | null>, key: string) => {
-              result[key] = selectedFilters[key].length ? selectedFilters[key].join(',') : null;
-
-              return result;
-            }, {});
-
-          this.setRouteParams({
-            ...this.router.routerState.root.snapshot.queryParams,
-            ...filterParams,
-          });
-        }),
-      );
-  }, { dispatch: false });
-
-  searching$ = createEffect(() => {
-    return this.actions$
-      .pipe(
-        ofType(FilterActions.setSearch),
-        tap(({ searchValue }) => {
-          this.setRouteParams({
-            ...this.router.routerState.root.snapshot.queryParams,
-            searchQuery: searchValue || null,
-          });
-        }),
-      );
-  }, { dispatch: false });
-
-  sorting$ = createEffect(() => {
-    return this.actions$
-      .pipe(
-        ofType(FilterActions.setSortingOption),
-        tap(({ sortingOption }) => {
-          this.setRouteParams({
-            ...this.router.routerState.root.snapshot.queryParams,
-            sortingOption,
-          });
-        }),
-      );
-  }, { dispatch: false });
+  updateRouteOnFilterChange$ = createEffect(() =>
+    this.router.events.pipe(
+      filter((event) => event instanceof NavigationEnd),
+      filter(() => !/^\/products\/\d+$/.test(this.router.url)),
```

Comment:
Put regex to constants & give it the name that will help us to understand what it does